Rent.BusinessLogic/Services/Implementations/VehicleClientHistoryService.cs

```diff
@@ -39,25 +64,91 @@ public async Task<VehicleClientHistoryModel> GetByIdAsync(Guid id, CancellationT
 
     public async Task<VehicleClientHistoryModel> AddAsync(VehicleClientHistoryModel vchModel, CancellationToken cancellationToken)
     {
+        var vehicleConnection = _catalogueServiceConnection + vchModel.VehicleId;
+        var clientConnection = _userServiceConnection + vchModel.ClientId;
+
+        var totalRentDays = (vchModel.EndDate - vchModel.StartDate).TotalDays;
+
+        if (double.IsNegative(totalRentDays))
+            throw new BadRequestException(ExceptionMessages.EndDateLessThanStartDate(vchModel.EndDate, vchModel.StartDate));
+
+        var vehicle = await GetFromServiceAsModelAsync<VehicleModel>(vehicleConnection, cancellationToken);
+
+        if (vehicle.IsRented)
+            throw new BadRequestException(ExceptionMessages.VehicleIsRented(vehicle.Id));
+
+        var client = await GetFromServiceAsModelAsync<ClientModel>(clientConnection, cancellationToken);
+
+        if (client.IsRenting)
+            throw new BadRequestException(ExceptionMessages.UserIsRenting(client.Id));
+
+        AssignAsRented(totalRentDays, vehicle, client);
+
         var vch = vchModel.Adapt<VehicleClientHistoryEntity>();
 
         await repository.AddAsync(vch, cancellationToken);
 
         var newVchModel = vch.Adapt<VehicleClientHistoryModel>();
 
+        var vehicleResponse = await PutInServiceAsync(vehicleConnection, vehicle, cancellationToken);
+
+        if (!vehicleResponse.IsSuccessStatusCode)
+        {
+            await repository.RemoveAsync(vch, cancellationToken);
+
+            await ProcessExceptionAsync(vehicleResponse, vehicleConnection, cancellationToken);
+        }
+
+        var userResponse = await PutInServiceAsync(clientConnection, client, cancellationToken);
+
+        if (!userResponse.IsSuccessStatusCode)
+        {
+            await repository.RemoveAsync(vch, cancellationToken);
+            await DeleteFromServiceAsync(clientConnection, cancellationToken);
+
+            await ProcessExceptionAsync(userResponse, _userServiceConnection, cancellationToken);
```

Comment:
Don't you think it is not a good flow to add and remove data?
What if removing data will fail?
Maybe require another approach